src/Chapter2.hs

```diff
@@ -605,8 +613,8 @@ Implement a function that duplicates each element of the list
 
 -}
 duplicate :: [a] -> [a]
-duplicate = error "duplicate: Not implemented!"
-
+duplicate [] = []
+duplicate (x:xs) = [x,x] ++ duplicate xs
```

Comment:
It is always better to use `:` instead of `++`, as it would be much more efficient 🙂 And here it is easily possible:

```suggestion
duplicate (x:xs) = x : x : duplicate xs
```